src/pools/abstract-pool.ts

```diff
@@ -311,6 +335,7 @@ export abstract class AbstractPool<
    * @returns Worker.
    */
   protected chooseWorker (): Worker {
+    this.checkEmitBusy()
```

Comment:
The entire checkEmitBusy method could have a performance overhead IMO.
We should at least have an implementation for each type of pool ( static,dynamic ).
One thing that I would consider instead is to check the number of current tasks ( promiseMap.size() ) and if is greater than the pool size ( min or max parameters ) than I would emit an event ( to discuss more details on that).
Another thing that I would do is define a maxConcurrentTasks option and then check if this number is passed, if yes then we can emit another event here ( still using the promiseMap.size(), and also take optional backpressure actions on it directly on poolifier ( i.e. backpressure delay, delay each task to X ms later without any impact on the user that still receive a promise ) .